### PR TITLE
Added UndoWithImagePlugin and updated ImageManageOptions' uploadImage to be File based instead of dataURL

### PR DESCRIPTION
**Pull request by adriantran**
Added UndoWithImagePlugin and updated ImageManageOptions' uploadImage to be File based instead of dataURL

### CODE REVIEW

File: packages/roosterjs-react-common/lib/plugins/UndoWithImagePlugin.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+     */
+    constructor(
+        private imageManager: ImageManagerInteface,
+        preserveSnapshots?: boolean,
+        private bufferSize: number = MAXSIZELIMIT
+    ) {
+        super(preserveSnapshots, bufferSize);
+    }
+
+    /**
```
Comment (pranavgk) [inline packages/roosterjs-react-common/lib/plugins/UndoWithImagePlugin.ts]: nit: not needed?

Comment (pranavgk) [reply packages/roosterjs-react-common/lib/plugins/UndoWithImagePlugin.ts]: I mean the whole dispose function here (as usual GitHub doesn't preserve highlighting :(

File: packages/roosterjs-react-common/lib/plugins/UndoWithImagePlugin.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        if (snapshot.hasPlaceholder) {
+            const originalValue = snapshot.value;
+            const result: UpdatePlaceholdersResult = this.imageManager.updatePlaceholders(originalValue);
+            snapshot.hasPlaceholder = !result.resolvedAll;
+            snapshot.value = result.html;
+            const delta = originalValue.length - result.html.length;
+
+            // it is possible total size is greater at this point (unlikely if default spinner is used)
+            this.totalSize -= delta;
+        }
```
Comment (pranavgk) [inline packages/roosterjs-react-common/lib/plugins/UndoWithImagePlugin.ts]: Can you add a comment describing this logic for future references?

File: packages/roosterjs-react-common/lib/plugins/UndoWithImagePlugin.ts (hunk 1 above)
Comment (adriantran) [reply packages/roosterjs-react-common/lib/plugins/UndoWithImagePlugin.ts]: deadcode

File: packages/roosterjs-react-common/lib/plugins/UndoWithImagePlugin.ts (hunk 2 above)
Comment (adriantran) [reply packages/roosterjs-react-common/lib/plugins/UndoWithImagePlugin.ts]: yeah I can provide a high level detail, there isn't much code to grasp

**Pull request merged**